pilot/pkg/config/coredatamodel/controller.go

```diff
@@ -240,9 +240,11 @@ func (c *Controller) HasSynced() bool {
 	c.syncedMu.Unlock()
 
 	if len(notReady) > 0 {
-		log.Infof("Configuration not synced: first push for %v not received", notReady)
+		time.Sleep(100 * time.Millisecond)
```

Comment:
This is not a good idea, if we want to inject delays we should do it in the waitForCacheSync, not here

Also removing the log entirely doesn't seem great either. Right now pilot is just going to loop forever with no message if you are missing a CRD.

Reply:
I can add it back - but right now it's printing that message in an loop without any delay or throttling, and you can't see much else.

Reply:
Also - you do see a line when the sync is completed (after 100ms ). Not sure a more complicated solution is needed, and there is little a user can do with 10000 lines 'not ready yet'.

Reply:
I am fine with removing the spam. What I don't like is that:
* It is NEVER logged. This makes it impossible to debug
* `HasSynced` sleeping 100ms. If we want to add throttling we should add that outside of this function

Reply:
It does log that sync is complete - do you need a log that it's still trying to sync, or it's starting
to sync ? What's the use case - what matters is when sync is complete ( or if it hangs - it'll be 
clear from the absence of the log ).

Reply:
The sleep is not for throttling - it's really bad to have
while(x) { log something } - waiting for x to change. It's also bad to do while(x) {}. 
I agree a more refined - channel, etc - could be used, but it's a startup only problem.

Reply:
In other words - no need to peg the CPU at 100% in a busy loop while galley is starting.

Reply:
I don't mean to be too picky about this, but the log line `"Configuration not synced: first push for %v not received"` has let me root cause issues at least 10 times. We need to be able to know what is not received, otherwise I don't know if I am missing all CRDs (galley is down?) or a single CRD (I didn't install correctly?).

If the pod is not getting ready but has no logs its not great. Pilot agent logs every 2s, that seems reasonable

Reply:
> In other words - no need to peg the CPU at 100% in a busy loop while galley is starting.

It was never doing a busy loop. It was looping with a 100ms interval, that is what `WaitForCacheSync`. Wait for 100ms is a huge hack. What if I want to call HasSynced as part of some different code (not startup)? now it arbitrarily waits 100ms

Reply:
1. If WaitForCacheSync can sleep 100ms, why can't we do it here ? 
2. The 100ms is in the k8s registry. The coredeatamodel doesn't seem to have the same - or I didn't see it. After k8s sync it's a busy loop. 

I removed it for now - to unblock the rest of the PR. Since I also removed galley in-process MCP - it shouldn't matter 
except for people who use MCP ( for them there is no regression - only change is we don't fill the logs in the busy loop). If you know of any sleep in that code path - then they won't have any problem.